### PR TITLE
Make the ESP32 Virtual Device Classes slightly easier to read

### PR DESCRIPTION
**Pull request by sagar-apple**
#### Problem
The virtual devices in the ESP32 example use some nested classes with single letter variable names. This is a little hard to read. 

 #### Summary of Changes
Try to make it easier to read by replacing single letter names with more descriptive names.

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

File: examples/all-clusters-app/esp32/main/main.cpp [hunk 1]
```diff
@@ -135,18 +135,22 @@ void AddDevice(std::string name)
 
 class EditAttributeListModel : public ListScreen::Model
 {
-    int d;
-    int e;
-    int c;
-    int a;
+    int endPointIndex;
```
Comment (mlepage-google) [inline examples/all-clusters-app/esp32/main/main.cpp:138]: Just note that d e c a stood for device endpoint cluster attribute
and the new names do not correspond.
Ensure that is what is intended.

Comment (sagar-apple) [reply examples/all-clusters-app/esp32/main/main.cpp:138]: doh! I will fix this. Thank you!

**Review by andy31415 — APPROVED** [review]

File: examples/all-clusters-app/esp32/main/main.cpp [hunk 2]
```diff
@@ -135,18 +135,22 @@ void AddDevice(std::string name)
...
+    EditAttributeListModel(int endPointIndex, int clusterIndex, int attributeListIndex, int attributeIndex) :
+        endPointIndex(endPointIndex), clusterIndex(clusterIndex), attributeListIndex(attributeListIndex),
+        attributeIndex(attributeIndex)
+    {}
     virtual std::string GetTitle()
     {
-        auto & attribute = std::get<1>(std::get<1>(std::get<1>(devices[d])[e])[c])[a];
-        auto & name      = std::get<0>(attribute);
-        auto & value     = std::get<1>(attribute);
+        auto & attribute =
```
[the remark this answers is not in the document]
Comment (sagar-apple) [reply examples/all-clusters-app/esp32/main/main.cpp:150]: Agreed. got another patching coming up for this.

File: examples/all-clusters-app/esp32/main/main.cpp [hunk 3]
```diff
@@ -135,16 +135,22 @@ void AddDevice(std::string name)
...
 class EditAttributeListModel : public ListScreen::Model
 {
-    int d;
-    int e;
-    int c;
-    int a;
+    int deviceIndex;
+    int endpointIndex;
+    int clusterIndex;
+    int attributeIndex;
```
Comment (sagar-apple) [inline examples/all-clusters-app/esp32/main/main.cpp:141]: @mlepage-google hopefully I got it right this time?

**Review by mspang — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**